explorer/src/pages/TransactionDetailsPage.tsx

```diff
@@ -410,6 +415,129 @@ function AccountsCard({
   );
 }
 
+type TokenBalanceRow = {
+  account: PublicKey;
+  mint: string;
+  balance: TokenAmount;
+  delta: number;
+  accountIndex: number;
+};
+
+function generateTokenBalanceRows(
+  preTokenBalances: TokenBalance[],
+  postTokenBalances: TokenBalance[],
+  accounts: ParsedMessageAccount[]
+): TokenBalanceRow[] {
+  let preBalanceMap: { [index: number]: TokenBalance } = {};
+
+  preTokenBalances.forEach(
+    (balance) => (preBalanceMap[balance.accountIndex] = balance)
+  );
+
+  return postTokenBalances
+    .map(
+      ({ uiTokenAmount, accountIndex, mint }): TokenBalanceRow => {
+        let delta;
+
+        if (accountIndex in preBalanceMap) {
+          delta = normalizeTokenDelta(
+            uiTokenAmount,
+            preBalanceMap[accountIndex].uiTokenAmount
+          );
+        } else {
+          delta = uiTokenAmount.uiAmount;
+        }
+
+        return {
+          account: accounts[accountIndex].pubkey,
+          mint,
+          balance: uiTokenAmount,
+          delta,
+          accountIndex,
+        };
+      }
+    )
+    .sort((a, b) => a.accountIndex - b.accountIndex);
+}
+
+function TokenBalancesCard({ signature }: SignatureProps) {
+  const details = useTransactionDetails(signature);
+  const { cluster } = useCluster();
+
+  if (!details) {
+    return null;
+  }
+
+  const preTokenBalances = details.data?.transaction?.meta?.preTokenBalances;
+  const postTokenBalances = details.data?.transaction?.meta?.postTokenBalances;
+  const accountKeys =
+    details.data?.transaction?.transaction.message.accountKeys;
+
+  if (!preTokenBalances || !postTokenBalances || !accountKeys) {
+    return null;
+  }
+
+  const rows = generateTokenBalanceRows(
+    preTokenBalances,
+    postTokenBalances,
+    accountKeys
+  );
+
+  if (rows.length < 1) {
+    return null;
+  }
+
+  const accountRows = rows.map(({ account, delta, balance, mint }) => {
+    const key = account.toBase58();
+    const renderChange = () => {
+      if (delta > 0) {
+        return <span className="badge badge-soft-success">+{delta}</span>;
+      } else if (delta < 0) {
+        return <span className="badge badge-soft-warning">{delta}</span>;
+      }
+      return <span className="badge badge-soft-secondary">{delta}</span>;
```

Comment:
Nice, do you mind adding the `0` state to how we display SOL delta as well? Maybe create a component that can be shared as well

Reply:
Good call.